be1-go/channel/coin/uint53/mod.go

```diff
@@ -0,0 +1,37 @@
+/*
+  Operations on 53-bits non-negative integers.
+
+  These values can be handled without loss of precision in IEEE754 64-bit
+  floating numbers, and thus in JSON.
+*/
+package uint53
+
+import (
+	"errors"
+)
+
+// Internal representation in this implementation
+type Uint53 = uint64
+
+// Maximum allowed value for Uint53
+const (
+	MaxUint53 Uint53 = 0x1F_FF_FF_FF_FF_FF_FF
+)
+
+// Check whether the value is in range
+func InRange(a Uint53) bool {
+	return a <= MaxUint53
+}
+
+// Compute the sum of two Uint53 values, or indicate an overflow if the sum is too large.
+func SafePlus(a, b Uint53) (r Uint53, err error) {
+	if !InRange(a) || !InRange(b) {
+		panic("Uint53.SafePlus: argument out of range")
```

Comment:
I would not panic and return an error. We probably don't want to server to crash for this reason.

Reply:
I look at this more as a violation of pre-conditions than something that goes wrong with the operation itself, that said it's the only panic in the codebase so I will change it.